src/internal/NeoEsp32I2sXMethod.h

```diff
@@ -292,6 +354,8 @@ template<typename T_SPEED, typename T_BUS, typename T_INVERT> class NeoEsp32I2sX
         Serial.println(_sizeData);
         
         _bus.RegisterNewMuxBus(_sizeData + T_SPEED::ResetTimeUs / T_SPEED::ByteSendTimeUs);
+        // data cleared later in Begin()
+        _data = static_cast<uint8_t*>(malloc(_sizeData));
```

Comment:
allocation should be avoided in the constructors here, this is what the initialize is for.  This is library mandate so in the next major version, initialize will return failure code for failed allocations.

Reply:
That fixed the issue I was talking with you on the discord: damaged LED data. Moving to the constructor fixed it. Later the DMA buffer size bug was discovered and maybe it was causing it since FillBuffer was running wild on the memory.

Reply:
OK, changes were reversed and the allocation was moved back to the initialize.